sw/host/hsmtool/src/commands/rsa/decrypt.rs

```diff
@@ -0,0 +1,59 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::{anyhow, Result};
+use cryptoki::context::Pkcs11;
+use cryptoki::mechanism::Mechanism;
+use cryptoki::object::Attribute;
+use cryptoki::session::Session;
+use serde::{Deserialize, Serialize};
+use serde_annotate::Annotate;
+use std::any::Any;
+use std::path::PathBuf;
+
+use crate::commands::{BasicResult, Dispatch};
+use crate::error::HsmError;
+use crate::util::attribute::KeyType;
+use crate::util::helper;
+
+#[derive(clap::Args, Debug, Serialize, Deserialize)]
+pub struct Decrypt {
+    #[arg(long)]
+    id: Option<String>,
+    #[arg(short, long)]
+    label: Option<String>,
+    #[arg(short, long)]
+    output: Option<PathBuf>,
+    input: PathBuf,
+}
+
+#[typetag::serde(name = "rsa-decrypt")]
+impl Dispatch for Decrypt {
+    fn run(
+        &self,
+        _context: &dyn Any,
+        _pkcs11: &Pkcs11,
+        session: Option<&Session>,
+    ) -> Result<Box<dyn Annotate>> {
+        let session = session.ok_or(HsmError::SessionRequired)?;
+        let mut attrs = helper::search_spec(self.id.as_deref(), self.label.as_deref())?;
+        attrs.push(Attribute::KeyType(KeyType::Rsa.try_into()?));
+        attrs.push(Attribute::Decrypt(true));
+
+        let mut object = session.find_objects(&attrs)?;
+        if object.is_empty() {
+            return Err(anyhow!("Found no matching key objects"));
+        } else if object.len() > 1 {
+            return Err(anyhow!("Found more than one matching key object"));
```

Comment:
Presumably this error happens when the user isn't providing enough information about the key they want to operate on. In that case, is there something that you can print about the found keys to help the user narrow it down further?
Given how many copies there are of this same call-`find_objects`-then-error-if-`len != 1` logic, maybe it should be in a helper function somewhere?

Reply:
I've refactored all of these into a single helper function.
- The first error case (is_empty) is usually an issue with specifying the key's `--id` or `--label` (e.g. a typo).
- The second error case (more than one) requires what I would consider to be an HSM misconfiguration: more than one key object with the same `label` or `id`.  I think this is technically possible, but probably rare in practice.

In each of the error cases, I've attached to the error message the search spec that led to the problem.